### DISCUSSION TITLE
Twitter like application

### DISCUSSION THREAD

**Discussion by jonas-martinez** [lenra-io/lenra_cli#54] [General]
To get Lenra started with its beta, the Lenra team wants to create a useful application on its store to showcase the power of Lenra on complex work. I'm opening this discussion to discuss with the team about how we want to implement this application and what type of work has to be done.

**Comment by jonas-martinez** [comment]
## Handling user profiles

### User table

- name
- identifier (like the @ identifier on twitter)
- picture? - defaulting to a placeholder profile image
- description?
- subscribers - [ ]
- subscribed_to - [ ]

CRUD operations must be implemented for this table.

### Tasks

- Implement CRUD operations for the user table. It could be useful to create a `User` object to have autocompletion. (with User->JSON and JSON->User methods)
- Implement methods to subscribe to a user

**Reply by jonas-martinez** [reply]
@lenra-io/devs I don't see anything else that needs to be done for a first version of the twitter like app, don't hesitate to comment to add more tasks or precisions

**Comment by jonas-martinez** [comment]
## Handling interaction between users

This feature will require the `Handling user profiles` feature. 
Users can interact with each other by sending private messages so we need to implement a chatting system. Nothing really out of the ordinary here.

Handling private messages should be enough for the moment and new features could be added in the future.

**Comment by jonas-martinez** [comment]
## Handling general thread

- A user can send a message on the application for everyone to see
- Users can like a message
- Users can comment on a message
- Users can repost the message (not really needed, more like a bonus)

With this, it should work properly.
We could add in the future a recommendation system (most liked messages, ...) or add the possibility to follow some specific topics.

**Reply by shiipou** [reply]
Did you think we can handle different types of timelines?

Like the following :
 - Recommendation Timeline
 - Trends Timeline
 - Subscription Timeline
 - Timeline of your own Profile (or a page to view your profile and the posts like a `public view` or something like that)

**Reply by jonas-martinez** [reply]
Yes this will need to be done just after the "initial" feature. 
For the moment, we just want to implement a working timeline with all the tweets ordered from newest to older ones.
